app/javascript/components/forms/CreateRoomForm.jsx

```diff
@@ -6,26 +6,25 @@ import {
 import PropTypes from 'prop-types';
 import Form from './Form';
 import Spinner from '../shared/stylings/Spinner';
-import useCreateRoom from '../../hooks/mutations/rooms/useCreateRoom';
 import FormControl from './FormControl';
 import { createRoomFormConfig, createRoomFormFields } from '../../helpers/forms/CreateRoomFormHelpers';
 
-export default function CreateRoomForm({ handleClose }) {
+export default function CreateRoomForm({ mutation: useCreateRoomAPI, handleClose }) {
+  const { mutate, isLoading } = useCreateRoomAPI({ onSettled: handleClose });
```

Comment:
I don't understand.
Should we keep using the wrappers (submit handler functions)?
But that is not compatible with the new style of mutation since we'll be returning something like `{handleSubmit, ...mutation}`.

Or, you just need me to keep using the `isSubmitting` state but without returning those wrappers so we change the `mutate` to `mutateAsync` instead?

Reply:
`const createRoomAPI = useCreateRoomAPI({ onSettled: handleClose });`